submit_cgap/submission.py

```diff
@@ -574,26 +645,62 @@ def submit_any_ingestion(ingestion_filename, *, ingestion_type, server, env, val
 
     uuid = res['submission_id']
 
-    show("Bundle uploaded. Checking ingestion process using IngestionSubmission uuid: %s ..." % uuid, with_time=True)
+    if DEBUG_PROTOCOL:  # pragma: no cover
+        show(f"Created IngestionSubmission object: s3://{metadata_bundles_bucket}/{uuid}", with_time=True)
+    show(f"Bundle uploaded to bucket {metadata_bundles_bucket}, assigned uuid {uuid} for tracking."
+         f" Awaiting processing...",
+         with_time=True)
+
+    check_done, check_status, check_response = check_submit_ingestion(uuid, server, env, app)
+
+    if validate_only:
+        exit(0)
+
+    if check_status == "success":
+        do_any_uploads(check_response, keydict=keydict, ingestion_filename=ingestion_filename,
+                       upload_folder=upload_folder, no_query=no_query,
+                       subfolders=subfolders)
+
+    exit(0)
+
+
+def _check_ingestion_progress(uuid, *, keypair, server) -> Tuple[bool, str, dict]:
+    """
+    Calls endpoint to get this status of the IngestionSubmission uuid (in outer scope);
+    this is used as an argument to check_repeatedly below to call over and over.
+    Returns tuple with: done-indicator (True or False), short-status (str), full-response (dict)
+    From outer scope: server, keypair, uuid (of IngestionSubmission)
+    """
+    tracking_url = ingestion_submission_item_url(server=server, uuid=uuid)
+    response = portal_request_get(tracking_url, auth=keypair, headers=STANDARD_HTTP_HEADERS).json()
+    # FYI this processing_status and its state, progress, outcome properties were ultimately set
+    # from within the ingester process, from within types.ingestion.SubmissionFolio.processing_status.
+    status = response.get("processing_status", {})
+    if status.get("state") == "done":
+        outcome = status.get("outcome")
+        return True, outcome, response
+    else:
+        progress = status.get("progress")
+        return False, progress, response
+
+
+def check_submit_ingestion(uuid: str, server: str, env: str,
+                           app: Optional[OrchestratedApp] = None) -> Tuple[bool, str, dict]:
```

Comment:
Type annotations would be great to have for other code in this PR as well.

Reply:
I'll look into adding some, though this is a work in progress.